tf-modules/distribution/main.tf

```diff
@@ -205,3 +212,44 @@ resource "aws_api_gateway_deployment" "s3_credentials" {
   rest_api_id = module.thin_egress_app.rest_api.id
   stage_name  = var.api_gateway_stage
 }
+
+# Egress Api Gateway Log Group Filter
+resource "aws_cloudwatch_log_subscription_filter" "egress_api_gateway_log_subscription_filter" {
+  count           = var.log_destination_arn != null && var.log_api_gateway_to_cloudwatch ? 1 : 0
```

Comment:
Why are most of the conditionals checking `var.log_destination_arn != null`, but this one is checking `var.log_destination_arn != null && var.log_api_gateway_to_cloudwatch`?

Reply:
That's a good question. I wouldn't ever expect `log_destination_arn` to be set without `log_api_gateway_to_cloudwatch`, but it is technically possible for a user to do try and run a deploy without both. This resource deals with _both_ log forwarding and API Gateway Logging.

For example, if a user were to want to propagate their egress lambda logs (which are created by default) but didn't want to enable API Gateway logging (for some reason), they could set a `log_destination_arn` without setting `log_api_gateway_to_cloudwatch`.